app/src/main/java/com/example/anfcodetest/MainActivity.kt

```diff
@@ -1,55 +1,36 @@
 package com.example.anfcodetest
 
-import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
 import androidx.recyclerview.widget.LinearLayoutManager
-import com.fasterxml.jackson.module.kotlin.*
+import com.hannesdorfmann.mosby3.mvp.MvpActivity
 import kotlinx.android.synthetic.main.activity_main.*
-import okhttp3.*
-import java.io.IOException
 
-val url = "https://www.abercrombie.com/anf/nativeapp/qa/codetest/codeTest_exploreData.json"
-
-class MainActivity : AppCompatActivity(), MainContract.View {
+class MainActivity : MvpActivity<MainContract.View, MainContract.Presenter>(), MainContract.View {
 
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
     setContentView(R.layout.activity_main)
-
     loadCards()
   }
 
+  override fun displayCards(cards: List<Card>) {
```

Comment:
We could add this method in to switch our adapter and load up our cards in our recycler view once they are ready. We don't need to iterate over the list anymore because the adapter and recycler views handle that for us!